_studio/shared/umc/codec/h264_dec/src/umc_h264_dec_bitstream_headers.cpp

```diff
@@ -373,15 +371,12 @@ inline bool CheckLevel(uint8_t level_idc, bool ignore_level_constrain = false)
     case H264VideoDecoderParams::H264_LEVEL_52:
 
     case H264VideoDecoderParams::H264_LEVEL_9:
-        return true;
 
-#if (MFX_VERSION >= 1035)
     case H264VideoDecoderParams::H264_LEVEL_6:
     case H264VideoDecoderParams::H264_LEVEL_61:
     case H264VideoDecoderParams::H264_LEVEL_62:
-        return ignore_level_constrain;
-#endif
-
+        return true;
```

Comment:
Why are you removing API level macro check? `(MFX_VERSION >= 1035`. Level 6.0 appeared in certain API level, you should not remove it.

Reply:
Code should be:
```
    case H264VideoDecoderParams::H264_LEVEL_51:
    case H264VideoDecoderParams::H264_LEVEL_52:

    case H264VideoDecoderParams::H264_LEVEL_9:

#if (MFX_VERSION >= 1035)
    case H264VideoDecoderParams::H264_LEVEL_6:
    case H264VideoDecoderParams::H264_LEVEL_61:
    case H264VideoDecoderParams::H264_LEVEL_62:
#endif
        return true;
```

in VPL we don't have MFX_VERSION macro because 2.x API series has this API from the very beginning. For MSDK  you must preserve MFX_VERSION protection.

Reply:
Your suggestion is reasonable, code changed